File: frontend/src/types/property.ts
export interface Property {
  id: string;
  name: string;
  address: string;
  timezone: string;
  aliases: string[];
  housekeepers_needed: number;
  is_active: boolean;
  metadata: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

export interface PropertyListItem {
  id: string;
  name: string;
  address: string;
  timezone: string;
  is_active: boolean;
  housekeepers_needed: number;
  property_manager: string | null;
}

export interface PropertySummary extends Property {
  active_incidents: number;
  items_in_laundry: number;
  low_stock_alerts: number;
  upcoming_reservations: number;
  current_guest: string | null;
  property_manager: string | null;
}

File: backend/app/domain/entities/property.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class Property(BaseModel):
    """A managed hospitality property (apartment, house, hotel, etc.).

    Each property has a canonical name, a physical address, and an
    IANA timezone used to localise timestamps for the operations team.
    Aliases allow the NLP layer to match informal property references
    (e.g. "el apto de la 93") to the canonical record.
    """

    id: UUID
    name: str
    address: str
    timezone: str = Field(default="America/Bogota")
    aliases: list[str] = Field(default_factory=list)
    housekeepers_needed: int = 1
    is_active: bool = True
    metadata: dict = Field(default_factory=dict)
    created_at: datetime
    updated_at: datetime
